### PR TITLE
Add Serious Sans font

### PR DESCRIPTION
**Pull request by kaBeech**
#### Description

This adds Serious Sans, a fork of Comic Shanns/Comic Mono I made that improves legibility, mainly by disambiguating the 'a'/'o' and '1'/'l' glyphs, and also adds Light and Italic styles

#### Requirements / Checklist

- [x] Read the [Contributing Guidelines](https://github.com/ryanoasis/nerd-fonts/blob/-/contributing.md)
- [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan.
      Issue number where discussion took place: #xxx
- [x] If this contains a font/glyph add its origin as background info below (e.g. URL)
- [x] Verified the license of any newly added font, glyph, or glyph set. License is: MIT

#### What does this Pull Request (PR) do?

Adds the Serious Sans font

#### How should this be manually tested?

Ideally, install it for your Terminal/IDE and check it out!

#### Any background context you can provide?

Font source is [Serious Sans on GitHub](https://github.com/kaBeech/serious-sans)

#### What are the relevant tickets (if any)?

N/A

#### Screenshots (if appropriate or helpful)

serious_sans_vsc

serious_sans_terminal

### CODE REVIEW

**Comment by kaBeech** [comment]
@all-contributors please add @kaBeech for code, design

**Comment by Finii** [comment]
Thanks @kaBeech for suggesting this interesting font!

Some thoughts on the font (superficial review) following.
See bottom for comments on the PR.

* Why fork and not contribute to some active Comic repo
* Charset very limited, no accents, ... 🤔 
* Just checked _Bold_, which seems autogenerated and no tweaking afterwards? Tons of self-intersections 😞 [1]
* `Serious Sans` as name is already taken, as you point out, not only `Serious Sans Pro` - and with a font of similar characteristics [2]
* The version number of the font is strange (forgotten to set?) [3]

### References

Self intersections everywhere as usual when using for example `fontforge`'s make bolder function. The intersections are tiny but ultimately can cause problems with some font renderers. Missing points at extrema is usually not a bigger problem nowadays.

On the other hand it validates fine in FontBook and Glyphs3.

Screenshot 2024-06-06 at 08 05 19

_Reference 1_

Screenshot 2024-06-06 at 08 11 21

_Reference 2_

Screenshot 2024-06-06 at 08 35 03

_Reference 3_

-----

### Regarding the inclusion of Serious Sans to the prepatched Nerd Font set

Although it would "solve" #1393 I am not sure the added weights and styles (Light. Bold, Italic) are hand optimised or just script runs.
Furthermore, before Comic Shanns had been included in Nerd Fonts all the free repos of Comic Sans descendants had been checked and Shanns had been chosen as the most active and advanced fork. Going back to a Comic Mono fork (which is a fork of Comic Shanns with just some script run added) feels very strange.

Screenshot 2024-06-06 at 09 01 12

_Serious Sans forked Comic Mono, a stale Comic Shanns fork?_

I believe this could be a good addition to Comic Shanns Mono instead of a stand alone half-top-of-the-lineage fork. If we add this font to NF users will get some more weights compared to Comic Shanns, but loose all the other good stuff. I literally see Issues pouring in here ;-)

So, while I believe the Italics (to get a full RIBBI set) and more weights (not so much) would be a good addition, this is not really cutting it.

I will add it to the suggested font list in #1095 and close this for now here.

Thank you again!

**Review by Finii — COMMENTED** [review]
Some technical comments for the PR, that might need to be addressed before pulling.

All other aspects seem to be filled out extremely careful (i.e. correct)! 👍

**Comment by kaBeech** [comment]
Hey @Finii, thanks so much for the review!

> * Why fork and not contribute to some active Comic repo

To be perfectly honest, this came down to 3 main things:

    - I wanted to feel free to mangle the font and make it my own without worrying about sending back a bunch of messy commits or accidentally messing something up upstream
    - At the time I judged the two most widely used monospaced Comic Sans descendants as Comic Shanns (which seemed/seems inactive), and Comic Mono. I forked Comic Mono because it seemed more well used and fixed some monospacing issues (looking back, trivial to fix, but I was a complete neophyte at the time) and I seem to remember doing something that I didn't want to override dtinth's work, but now I'm not sure what it was. I do not know why the fantastic Comic Shanns Mono escaped my attention - I feel like I must have seen [this PR](https://github.com/shannpersand/comic-shanns/pull/19) while researching
    - I was really tickled about the name

> * Charset very limited, no accents, ... 🤔

Very true. I remember vaguely wondering about this but I didn't notice any issues when coding in Spanish or Arabic, so I didn't worry too much about it. Looking at my code now I can see that the missing chars were substituted in from other fonts 😅

> * Just checked _Bold_, which seems autogenerated and no tweaking afterwards? Tons of self-intersections 😞

This is probably correct (obviously correct about the self-intersections). I did autogenerate the Bold/Light/Italic styles, and I remember tweaking the Light and Italic styles, but only for the way it looked visually. It's entirely possible that I didn't tweak Bold at all. Honestly I'm still a font n00b and don't know much about what issues need to be fixed vs what is less necessary to address

> * `Serious Sans` as name is already taken, as you point out, not only `Serious Sans Pro` - and with a font of similar characteristics

I was actually confused about this when originally working on this. I saw that also but I didn't find any actual `Serious Sans` font that wasn't `Serious Sans Pro`. Point taken though.

> * The version number of the font is strange (forgotten to set?)

Oop, yep, thanks for catching that!

> ### Regarding the inclusion of Serious Sans to the prepatched Nerd Font set...

All told, I'll look into submitting a PR to Comic Shanns Mono with my modified 'a' and 'l' glyphs. These glyph modifications are main advantage to me for using Serious Sans compared with alternatives. I like having Italic versions available, but I remember getting them to look good taking way more work than I have time for at the moment and I feel like submitting a PR that adds Italics for the limited charset without touching the other chars might be tacky

If my PR to Comic Shanns Mono gets declined I will adjust my font to address these issues and then touch back in. In case I do go that direction, would you be able to offer an opinion as to whether the font would be acceptable under the name `Serious Sans Mono`?

> Thank you again!

Hey friend, I appreciate all the work you're putting into this project! Nerd Fonts has been extremely helpful for me, both the official fonts and the Font Patcher. It's helped me feel empowered to adapt the tools I use for my specific needs. And I really appreciate the quick and thorough PR response! Thank you!

File: bin/scripts/lib/fonts.json [hunk 1]
```diff
@@ -783,6 +783,19 @@
...
       "description": "Dashed zero, curved and straight character lines"
+    },    {
+      "unpatchedName": "Serious Sans",
+      "licenseId": "MIT",
+      "RFN": false,
+      "version": "3.0.1",
+      "patchedName": "SeriousSans",
+      "folderName": "SeriousSans",
+      "imagePreviewFont": "SeriousSans Nerd Font",
+      "imagePreviewFontSource": "SeriousSans/Regular/SeriousSans-Regular.otf",
```
[the remark this answers is not in the document]
Comment (kaBeech) [reply bin/scripts/lib/fonts.json:794]: Oop, yep, my mistake, thanks!

File: patched-fonts/SeriousSans/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Serious Sans
```
[the remark this answers is not in the document]
Comment (kaBeech) [reply patched-fonts/SeriousSans/README.md:1]: Could you please clarify what you mean here? Do you mean that we should not commit the autogenerated patched font files to commits that add new fonts?

File: src/unpatched-fonts/SeriousSans/config.cfg [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[Config]
+commandline: --makegroups 3
```
[the remark this answers is not in the document]
Comment (kaBeech) [reply src/unpatched-fonts/SeriousSans/config.cfg:2]: 🙌

File: patched-fonts/SeriousSans/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kaBeech) [reply patched-fonts/SeriousSans/README.md:1]: @Finii If this is the intention, I have [this commit](https://github.com/ryanoasis/nerd-fonts/compare/master...kaBeech:nerd-fonts:clarify-documentation#diff) that updates the documentation to make that clearer. If it is helpful I'd be happy to submit a PR with this change 😸

File: bin/scripts/lib/fonts.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (kaBeech) [reply bin/scripts/lib/fonts.json:794]: Fixed

File: patched-fonts/SeriousSans/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kaBeech) [reply patched-fonts/SeriousSans/README.md:1]: Also fixed 😸

[the remark this answers is not in the document]
Comment (kaBeech) [reply patched-fonts/SeriousSans/README.md:1]: Moved this question to #1655 - resolving this conversation

**Pull request closed without merging**